File: fix_urls.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(file));
        } else if (file.endsWith('.js') || file.endsWith('.jsx')) {
            results.push(file);
        }
    });
    return results;
}

const files = walk('frontend/src');
let count = 0;

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    const regex = /import\.meta\.env\.VITE_API_URL\s*\|\|\s*(["'])http:\/\/localhost:5000\1(?!\)\.replace)/g;
    
    if (regex.test(content)) {
        content = content.replace(regex, '(import.meta.env.VITE_API_URL || $1http://localhost:5000$1).replace(/\\\\/$/, "")');
        fs.writeFileSync(file, content, 'utf8');
        console.log('Updated:', file);
        count++;
    }
});
console.log('Total files updated:', count);

File: fix_urls.py
import os
import glob
import re

for filepath in glob.glob('frontend/src/**/*.js*', recursive=True):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    pattern = r"import\.meta\.env\.VITE_API_URL\s*\|\|\s*([\"'])http://localhost:5000\1(?!\)\.replace)"
    
    if re.search(pattern, content):
        new_content = re.sub(pattern, r'(import.meta.env.VITE_API_URL || \1http://localhost:5000\1).replace(/\/$/, "")', content)
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print('Updated:', filepath)
